### ISSUE TITLE
Add support for `two connectors one buffer`

### ISSUE THREAD

**Issue by TheNeikos** [waysome/waysome#175]
Exactly what it says.

**Comment by matthiasbeyer** [comment]
Q: @TheNeikos Can you provide a list of steps what has to be done for this?

**Comment by TheNeikos** [comment]
One would have to abstract the framebuffer of a monitor in such a way that it could also point to an already existing framebuffer.

```

      Monitor ------ Buffer -----
                                            |-------Real Buffer
      Monitor ------ Buffer -----
```

**Comment by matthiasbeyer** [comment]
So one has only to change the `ws_monitor` type to be able to get a `ws_frame_buffer` (the `buffer` member) set?

**Comment by TheNeikos** [comment]
No, each monitor still needs their own framebuffer as this is what the graphic card sends to the monitor.

Something like "This monitor copies exactly what is written in that other monitor" _everytime_ the other one updates.

**Comment by matthiasbeyer** [comment]
Hm. I understand what you mean, but I don't understand what needs to be done for this. I assume (although I cannot find the appropriate code) that each monitor object has something like a "in" buffer and a "out" buffer, and you want me to provide a possibility to make one monitors "in" buffer the "out" buffer of another one.

So I guess I have to look at `src/compositor/monitor.c`. But I don't see the appropriate code for this.

---

Oh man, I really have to get me into this stuff!

**Comment by TheNeikos** [comment]
Nope, there is no intermediary buffer right now, which is the 'problem'. Everything gets blitted right away!

**Comment by matthiasbeyer** [comment]
Hm. But blitting right away sounds much more efficient from why I understand, right? So it would be a change in the blitting code, to blit on several buffers at once instead of blitting in a temporary buffer, am I right?

**Comment by TheNeikos** [comment]
We already blit on all monitors at the same time ^^

We will first have to decide how to handle surface commits, right now they are still unsuitable to be used. I still suggest you read into the code, but not yet try to think on how to implement this.
